### PR TITLE
feat: add fuzz match for error 'attribute not found'

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y

fix issue #670  

#### 2. What is the scope of this PR (e.g. component or file name):

kclvm/sema/src/resolver/attr.rs

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

### CODE REVIEW

File: kclvm/sema/src/resolver/attr.rs [hunk 1]
```diff
@@ -110,16 +112,32 @@ impl<'ctx> Resolver<'ctx> {
                 }
             }
         };
+
         if !result {
+            let mut suggestion = String::new();
```
Comment (Peefy) [inline kclvm/sema/src/resolver/attr.rs:117]: Use a small function to get the closets list.

File: kclvm/utils/src/str.rs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+/// let input = "hello".to_string();
+/// let pools = vec!["hello".to_string(), "world".to_string()];
+/// let closest = find_closest_strs(input.to_string(), pools, None);
+/// assert_eq!(closest, vec!["hello"]);
+///
+/// let pools = vec!["good".to_string(), "world".to_string()];
+/// let closest = find_closest_strs(input, pools, None);
+/// assert_eq!(closest, Vec::<String>::new());
+/// ```
+pub fn find_closest_strs(
```
Comment (Peefy) [inline kclvm/utils/src/str.rs:20]: Why dont use a 3rd party lib.

File: kclvm/sema/src/resolver/attr.rs [hunk 3]
```diff
@@ -110,16 +112,32 @@ impl<'ctx> Resolver<'ctx> {
...
+            // Calculate the closests miss attributes.
+            if let Schema(schema_ty) = &obj.kind {
+                // Get all the attrbuets of the schema.
+                let attrs = schema_ty.attrs.keys().cloned().collect::<Vec<String>>();
+                // The attr user input.
+                let attr = if attr.is_empty() {
+                    "[missing name]"
```
Comment (Peefy) [inline kclvm/sema/src/resolver/attr.rs:124]: Missing name does not require the nearest attribute.

File: kclvm/cmd/src/tests.rs [hunk 4]
```diff
@@ -559,3 +559,39 @@ fn test_plugin_not_found() {
         Err(msg) => assert!(msg.contains("the plugin package `kcl_plugin.not_exist` is not found, please confirm if plugin mode is enabled")),
     }
 }
+
+#[test]
+fn test_error_message_fuzz_matched() {
```
Comment (Peefy) [inline kclvm/cmd/src/tests.rs:564]: Move these test cases into the resolver crate `kcl/kclvm/sema/src/resolver/tests.rs'`

**Review by Peefy — APPROVED** [review]
LGTM

**Review by i-zhen — APPROVED** [review]
LGTM

**Pull request merged**